test.ts

```diff
@@ -190,3 +190,66 @@ Deno.test("validateRequest() populates body as per schema", async () => {
   assertEquals(error, undefined);
   assertEquals(body, { name: "Satya", age: 98 });
 });
+
+const headersInitCases: {
+  description: string;
+  headers: HeadersInit;
+  entries: [string, string][];
+}[] = [
+  {
+    description: "merges Headers",
+    headers: new Headers({
+      "content-type": "type/subtype",
+      "custom-header-1": "1",
+      "custom-header-2": "2",
+    }),
+    entries: [
+      ["content-type", "type/subtype"],
+      ["custom-header-1", "1"],
+      ["custom-header-2", "2"],
+    ],
+  },
+  {
+    description: "merges [string, string][]",
+    headers: [
+      ["content-type", "type/subtype"],
+      ["custom-header-1", "1"],
+      ["custom-header-2", "2"],
+    ],
+    entries: [
+      ["content-type", "type/subtype"],
+      ["custom-header-1", "1"],
+      ["custom-header-2", "2"],
+    ],
+  },
+  {
+    description: "merges Record<string, string>",
+    headers: {
+      "content-type": "type/subtype",
+      "custom-header-1": "1",
+      "custom-header-2": "2",
+    },
+    entries: [
+      ["content-type", "type/subtype"],
+      ["custom-header-1", "1"],
+      ["custom-header-2", "2"],
+    ],
+  },
+];
+
+for (const { entries, headers, description } of headersInitCases) {
```

Comment:
Nice :)

Reply:
@satyarohith This can be extended for [`newResponse`](https://github.com/satyarohith/sift/blob/89a293c7f93fb9fda0bee4c9d147f376b0006bac/mod.ts#L60-L85) if you want to export it for testing. However, I didn't do that for a couple of reasons:

1. I didn't want to modify the API of the existing exports.
2. From looking at the code and comments, it seems like that function is only meant to be used internally to fulfill a temporary issue with deno deploy.